### PR TITLE
APP-366 captcha v2 improvements

### PR DESCRIPTION
**Pull request by yaronpx**

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -33,13 +33,8 @@ http {
...
             index  index.html;
-        }
-        location ~* \.(?:ico|css|js|gif|jpe?g|png|woff|ttf)$ {
-        #--- Some basic cache-control for static files to be sent to the browser
-            expires max;
-            add_header Pragma public;
-            add_header Cache-Control "public, must-revalidate, proxy-revalidate";
-            root   /nginx/html;
+
+            include  /nginx/conf/mime.types;
```
Comment (nitzanpx) [inline nginx.conf]: why was that changed?

File: lib/px/block/pxtemplate.lua [hunk 2]
```diff
@@ -111,6 +64,15 @@ function M.load(config_file)
         return content
     end
 
+    local function get_script(template, px_config)
```
Comment (nitzanpx) [inline lib/px/block/pxtemplate.lua]: I suggest calling this function inside `get_props`
https://github.com/PerimeterX/perimeterx-nginx-plugin/blob/44bdd6197d5c7ac844a9d8ae609aaed5093e5e0d/lib/px/block/pxtemplate.lua#L78

and calling `get_props` on line #81
https://github.com/PerimeterX/perimeterx-nginx-plugin/blob/44bdd6197d5c7ac844a9d8ae609aaed5093e5e0d/lib/px/block/pxtemplate.lua#L81

File: lib/px/utils/pxclient.lua [hunk 3]
```diff
@@ -219,6 +219,22 @@ function M.load(config_file)
...
+            ngx.exit(ngx.OK)
+            return
+        end
+        local reverse_prefix = string.sub(px_config.px_appId, 3, string.len(px_config.px_appId))
+        local px_request_uri = string.gsub(ngx.var.uri, '/' .. reverse_prefix .. px_constants.FIRST_PARTY_CAPTCHA_PATH, '')
+        px_logger.debug("Forwarding request from "  .. ngx.var.request_uri .. " to px captcha at " .. px_config.captcha_script_host .. px_request_uri)
+        ngx_req_set_uri(px_request_uri)
+
+        px_common_utils.clear_first_party_sensitive_headers(px_config.sensitive_headers);
+        _M.forward_to_perimeterx(px_config.captcha_script_host)
```
Comment (nitzanpx) [inline lib/px/utils/pxclient.lua]: need to add here port overide so we could pass a different port, not for clients use usually but its better that we have this under control

File: nginx.conf (hunk 1 above)
Comment (yaronpx) [reply nginx.conf]: couldn't make first party work locally without it

File: lib/px/utils/pxclient.lua (hunk 3 above)
Comment (yaronpx) [reply lib/px/utils/pxclient.lua]: I disagree. It's fine that `forward_to_perimeterx` accepts an override port because it makes it more generic, but for this flow I don't think there's a reason it would be configurable. we should also consider that a large number of config values is confusing for customers and potential developers
